src/pages/admin/settings.js

```diff
@@ -0,0 +1,107 @@
+import { Typography, TextField, Button, Snackbar, Box } from "@mui/material";
+import { gql, useQuery, useMutation } from "@apollo/client";
+import { useState } from "react";
+
+const QUERY = gql`
+    query {
+        settings {
+            membershipRequirement
+        }
+    }
+`;
+
+const MUTATION = gql`
+	mutation ($membershipRequirement: Int!) {
+        alterSettings(membershipRequirement: $membershipRequirement) {
+            membershipRequirement
+        }
+	}
+`;
+
+const classes = {
+	mainDiv: {
+		width: "1200px",
+		maxWidth: "90%",
+		margin: "auto"
+	}
+};
+
+const ManageSettings = () => {
+    const {
+		_loading,
+		_error,
+		data
+	} = useQuery(QUERY);
+
+    let [settings, setSettings] = useState(null)
+    let [changed, setChanged] = useState(false)
+
+    const [save] = useMutation(MUTATION, {
+        onCompleted() {
+            setChanged(false);
+        }
+    });
+
+    if (data && settings === null) {
+        setSettings(data.settings)
+    }
+
+    const saveSettings = async () => {
+        const confirmation = window.confirm("Are you sure you want to update these settings?");
+		if (!confirmation) {
+			return;
+		}
+
+        if (isNaN(settings.membershipRequirement)) {
+            alert("Error: Invalid Membership Requirements")
+            return;
+        }
+        
+        let variables = { membershipRequirement: settings.membershipRequirement}
+
+		try {
+			await save({ variables });
+			alert("Settings updated!");
+		} catch (e) {
+			alert("Error updating settings: " + e.message);
+		}
+    }
+
+    return (
+        <Box sx={classes.mainDiv}>
+            <Typography variant={"h3"}>StuyActivities Settings:</Typography>
+            <TextField 
+                label="Member Requirement" 
+                variant="outlined" 
+                value={settings === null ? "" : settings.membershipRequirement} 
+                onChange={e => {
+                    if (!e.target.value.length) {
+                        setSettings({ ...settings, membershipRequirement: "" })
+                        setChanged(false);
+                        return;
+                    }
+
+                    let newValue = parseInt(e.target.value)
+
+                    if (isNaN(newValue)) return;
```

Comment:
See previous comment

Reply:
In addition, the [field itself can complain on bad input](https://mui.com/material-ui/react-text-field/#validation)